lib/ae_mdw/db/origin.ex

```diff
@@ -143,21 +143,12 @@ defmodule AeMdw.Db.Origin do
   end
 
   defp hc_contracts do
-    with {:ok, {hardfork_key, _hf_val}} <-
-           :aeu_env.find_config(["chain", "hard_forks"], [
-             :user_config,
-             :schema_default,
-             {:value, {@iris_protocol, 0}}
-           ]),
-         {protocol, ""} <- Integer.parse(hardfork_key),
-         {:ok, hc_contracts_json} <-
-           :aec_fork_block_settings.hc_seed_contracts(protocol, :aec_governance.get_network_id()) do
-      hc_contracts_json
-      |> Enum.find_value(fn {item, list} -> if item == "contracts", do: list end)
-      |> Enum.map(&(Map.get(&1, "pubkey") || Map.fetch!(&1, "contract_pubkey")))
-      |> Enum.map(&Validate.id!/1)
-    else
-      _error_missing -> []
-    end
+    {:ok, hc_contracts_json} =
+      :aec_fork_block_settings.hc_seed_contracts(@iris_protocol, :aec_governance.get_network_id())
+
+    hc_contracts_json
+    |> Map.get("contracts", [])
```

Comment:
During development json that defines the hc seed was also defined as a list. This is the reason for the existing code that handles both maps and keywords with:
```
hc_contracts_json
|> Enum.find_value(fn {item, list} -> if item == "contracts", do: list end)
```
In other words it covers what `|> Map.get("contracts", [])` does and handles keywords as well.

Reply:
`hc_seed_contracts(protocol, network_id)` is always going to return the output of JSON decoding: https://github.com/aeternity/aeternity/blob/v6.12.0/apps/aecore/src/aec_fork_block_settings.erl#L358 which is never going to return keyword lists, since JSON maps are returned as erlang maps

Reply:
`Enum.find_value` still handles the decoding of any JSON which might be a map or a list and does not depend on the implementation.